docs: describe intended ways to use Renovate (#11197)
renovatebot/renovate: docs/development/design-decisions.md

This file documents the design choices as well as configuration options.

> ## Intended usage by end-users

Author: ```suggestion
## Intended use by end-users
```

> Renovate is intended to be used as an app.

Reviewer: ```suggestion
The Renovate repository/package is intended to be used as a CLI-based application.
```

> It should not be used as a library, because it lacks a stable API.

Reviewer: ```suggestion
It should not be used downstream as a library, because it lacks a stable API.
```

> We do not want to create a stable API, or document our existing API.

Reviewer 2: We want a stable api some time, but currently it's changing too often to be stable soon.

Author: Do we want to make any promises/future work by suggesting we're going to stabilize our API? 

@rarkins What do you think?

Author: ```suggestion
Right now we do not have a stable API.
```

> 
> Right now we do not have a stable API.
> The API and documentation would need a major overhaul each time we refactor our code.

Author: ```suggestion
```

We can probably get rid of these 3 lines now?
